google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/apis/CloudApiManagementConfirmationDialog.java

```diff
@@ -81,4 +138,110 @@ private void populateLibraryList(JList<String> list, Set<CloudLibrary> libraries
     libraries.forEach(library -> listModel.addElement(library.getName()));
     list.setModel(listModel);
   }
+
+  private void createUIComponents() {
+    roleTable = new ServiceAccountRolesTable(roles);
+  }
+
+  private static final class ServiceAccountRolesTable extends JBTable {
+    ServiceAccountRolesTable(Set<Role> roles) {
+      super(new ServiceAccountRolesTableModel(roles));
+      setDefaultRenderer(Role.class, new RoleNameRenderer());
+      setDefaultRenderer(Boolean.class, new BooleanTableCellRenderer());
+      setDefaultEditor(Boolean.class, new BooleanTableCellEditor());
+      TableUtil.setupCheckboxColumn(getColumnModel().getColumn(1));
+      setBorder(IdeBorderFactory.createBorder());
+      setRowHeight(25);
+    }
+
+    /** See {@link com.intellij.ide.plugins.PluginTable#paint(Graphics)} for reasoning. */
+    @Override
+    public void paint(@NotNull Graphics g) {
+      super.paint(g);
+      UIUtil.fixOSXEditorBackground(this);
+    }
+  }
+
+  private static final class RoleNameRenderer extends DefaultTableCellRenderer {
+
+    @Override
+    public Component getTableCellRendererComponent(
+        JTable table, Object role, boolean isSelected, boolean hasFocus, int row, int column) {
+      super.getTableCellRendererComponent(table, role, isSelected, hasFocus, row, column);
+      setBorder(BorderFactory.createEmptyBorder(5, 5, 5, 5));
+      setText(((Role) role).getTitle());
+      return this;
+    }
+  }
+
+  private static final class ServiceAccountRolesTableModel extends AbstractTableModel {
+
+    private final SortedMap<Role, Boolean> roleMap =
+        new TreeMap<>(Comparator.comparing(Role::getName));
+
+    ServiceAccountRolesTableModel(Set<Role> roles) {
+      roleMap.putAll(Maps.toMap(roles, role -> true));
+    }
+
+    @Override
+    public int getRowCount() {
+      return roleMap.size();
+    }
+
+    @Override
+    public int getColumnCount() {
+      return 2;
+    }
+
+    @Override
+    public Class<?> getColumnClass(int columnIndex) {
+      return columnIndex == 0 ? Role.class : Boolean.class;
+    }
+
+    @Override
+    public boolean isCellEditable(int rowIndex, int columnIndex) {
+      return columnIndex == 1;
+    }
+
+    @Override
+    public Object getValueAt(int rowIndex, int columnIndex) {
+      if (columnIndex == 0) {
+        return roleMap.keySet().toArray()[rowIndex];
+      }
+      return roleMap.values().toArray()[rowIndex];
+    }
+
+    @Override
+    public void setValueAt(Object value, int rowIndex, int columnIndex) {
+      if (columnIndex == 0) {
+        throw new RuntimeException();
+      }
+
+      Role role = (Role) roleMap.keySet().toArray()[rowIndex];
+      roleMap.put(role, (boolean) value);
+
+      TableModelEvent event = new TableModelEvent(this, rowIndex);
+      Stream.of(listenerList.getListeners(TableModelListener.class))
+          .forEach(listener -> listener.tableChanged(event));
```

Comment:
It might be more complicated than in needs to - it's possible table will update this cell itself after editing and event is not needed at all? Or if it is needed, there are a set of useful methods in `AbstractTableModel` -
 we can try `fireTableCellUpdated(rowIndex, columnIndex);`, it could be easier to read.

Reply:
I don’t think it is needed. I’ll check. Thanks